### ISSUE TITLE
Spyder 3 as an app on Mac

### ISSUE THREAD

**Issue by gkvallis** [spyder-ide/spyder#3595]
This is not so much an issue as a request for information. If I should sent it elsewhere please let me know.

I see that Spyder 3 is now available as a Mac app with a dmg download. I've previously installed and updated it with condo, but generally, on a Mac, I prefer to use a real app as you can switch to it, or kill (force quit) it, etc, using Mac tools. But with the Mac app the command line tools (like opening a new instance) seem more limited than the one in  ~/anaconda/bin.  Also, does the app store presences in the same place (i.e. .spyder-py3)? Can I use both the app and the version in ~/anaconda/bin?

In summary, what are the differences and pros/cons between the two versions?

Thanks
Geoff

**Comment by ccordoba12** [comment]
> But with the Mac app the command line tools (like opening a new instance) seem more limited than the one in ~/anaconda/bin

Well, you can't open two instances of an application in Mac, so we don't allow that for our app either :-)

> does the app store presences in the same place (i.e. .spyder-py3)?

Yes, it does.

> Can I use both the app and the version in ~/anaconda/bin?

Yep, you can do that without problems :-)

> what are the differences and pros/cons between the two versions?

1. Our app is completely self contained, i.e. you can't update its contents nor change its Python version.
2. The app gives you an icon in Mac's dock and also a proper name in the menu bar.
3. You can do a right-click on several file types on Finder (e.g. Python files) and open them directly in the Spyder app.
4. Our app doesn't use optimized (i.e. mkl enabled) versions of Numpy, Pandas, etc. It's purpose is to facilitate installation, mostly for Python courses that use Spyder. Although you can also use it (as many people does :-) to start Spyder as a proper app, but using your anaconda installation to run computations. For that you need to change your Python interpreter to point to `~/anaconda/bin/python` in

    Tools > Preferences > Python interpreter

----

Since this is not a proper Spyder bug, I'm closing this issue, but feel free to post more comments here if you have any doubts. I'll respond to them :-)

**Comment by gkvallis** [comment]
Thanks, that's very  helpful :) Answers my questions, except for a couple that may be different. 

1. When I start the app, then after switching the interpretor to ~/anaconda/bin using preferences (not in tools -> preferences btw)   it says I don't have the ipykernel installed, so it can't do anything, and it wants me to install it with conda or pip.  I will actually likely revert to use the unix version. 

2. When I go back to the ~/anaconda/bin version it gives me an error on startup:
   [warn] kq_init: detected broken kqueue; not using.: Undefined error: 0
and the kernel dies and restarts in the ipython console.   It works fine after that. This might be related to having moved to OS Sierra at  about the same time as I upgraded Spyder, not sure.

**Comment by ccordoba12** [comment]
> I will actually likely revert to use the unix version

I don't understand what you mean by this.

> [warn] kq_init: detected broken kqueue; not using.: Undefined error: 0

This seems to be a simple warning :-)

**Comment by gkvallis** [comment]
By unix version I just meant the version of Spyder  in ~/anaconda/bin, which works fine after the hiccup on starting. Not good terminology I know!

**Comment by ccordoba12** [comment]
> it says I don't have the ipykernel installed

Please be sure that you have `ipykernel` installed in your Anaconda installation.

> I will actually likely revert to use the unix version

Please select the default interpreter again and start the Anaconda Spyder instead of our app.

**Comment by mrclary** [comment]
Actually, I have the same ipykernel error on startup when selecting the Anaconda python interpreter. However, I _do_ have ipykernel installed...
